src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs

```
@@ -1687,14 +1677,21 @@ private PropertyTab CreateTab(Type tabType, IDesignerHost host)
                     parameter = Site;
                 }
 
-                if (parameter is not null && constructor is not null)
+                try
                 {
-                    tab = (PropertyTab)constructor.Invoke(new object[] { parameter });
+                    if (parameter is not null && constructor is not null)
+                    {
+                        tab = (PropertyTab)constructor.Invoke(new object[] { parameter });
+                    }
+                    else
+                    {
+                        // Just call the default constructor.
+                        tab = (PropertyTab)Activator.CreateInstance(tabType);
+                    }
                 }
-                else
+                catch
                 {
-                    // Just call the default constructor.
-                    tab = (PropertyTab)Activator.CreateInstance(tabType);
+                    tab = null;
```

This seems to be a silent failure. Consider adding either an assert or a `WriteLine`

---

It was a silent error before. I'll add a writeline.

---

Actually I'll make it an assert.